File: springframework/7.0.8/analyzed/spring-beans/src/main/java/org/springframework/beans/factory/annotation/Value.java
/*
 * Copyright 2002-present the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.springframework.beans.factory.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 可用于字段或方法/构造器参数，为被标注元素指定默认值表达式。
 *
 * <p>通常用于表达式驱动或属性驱动的依赖注入；也支持在处理器方法参数上动态解析，
 * 例如在 Spring MVC 中。
 *
 * <p>常见用法是通过 SpEL（Spring 表达式语言）注入值，例如
 * <code>#{systemProperties.myProp}</code>；也可使用
 * <code>${my.app.myProp}</code> 风格的属性占位符。
 *
 * <p>注意：{@code @Value} 的实际处理由
 * {@link org.springframework.beans.factory.config.BeanPostProcessor
 * BeanPostProcessor} 完成，因此<em>不能</em>在
 * {@link org.springframework.beans.factory.config.BeanPostProcessor
 * BeanPostProcessor} 或
 * {@link org.springframework.beans.factory.config.BeanFactoryPostProcessor BeanFactoryPostProcessor}
 * 类型中使用 {@code @Value}。请参阅 {@link AutowiredAnnotationBeanPostProcessor}
 * 的文档（默认会检测此注解是否存在）。
 *
 * @author Juergen Hoeller
 * @since 3.0
 * @see AutowiredAnnotationBeanPostProcessor
 * @see Autowired
 * @see org.springframework.beans.factory.config.BeanExpressionResolver
 * @see org.springframework.beans.factory.support.AutowireCandidateResolver#getSuggestedValue
 */
@Target({ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.ANNOTATION_TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Value {

	/**
	 * 实际值表达式，例如 <code>#{systemProperties.myProp}</code>，
	 * 或属性占位符，例如 <code>${my.app.myProp}</code>。
	 */
	String value();

}

File: prometheus/3.13.2/analyzed/promql/parser/value.go
// Copyright The Prometheus Authors
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// PromQL 求值结果类型定义：Value 接口与 instant/range vector、scalar、string 等 ValueType 常量及文档用语映射。

package parser

// Value 为查询求值结果的通用接口，提供类型标识与字符串化。
// Value is a generic interface for values resulting from a query evaluation.
type Value interface {
	Type() ValueType
	String() string
}

// ValueType 枚举 PromQL 表达式静态/动态值类别。
// ValueType describes a type of a value.
type ValueType string

// The valid value types.
const (
	ValueTypeNone   ValueType = "none"
	ValueTypeVector ValueType = "vector"
	ValueTypeScalar ValueType = "scalar"
	ValueTypeMatrix ValueType = "matrix"
	ValueTypeString ValueType = "string"
)

// DocumentedType 将内部 ValueType 映射为用户文档中的术语（如 instant vector）。
// DocumentedType returns the internal type to the equivalent
// user facing terminology as defined in the documentation.
func DocumentedType(t ValueType) string {
	switch t {
	case ValueTypeVector:
		return "instant vector"
	case ValueTypeMatrix:
		return "range vector"
	default:
		return string(t)
	}
}
